leasing/management/commands/import_level_check_index.py

```diff
@@ -0,0 +1,299 @@
+
+    Keys:
+        code: Identifier for the column.
+        text: Textual display name for the column.
+        type: Type of the column: "d" = dimension, "t" = time, "c" = measure. Optional. Default: "d".
+        unit: Unit for the measures. Only for type "c". Optional.
+        comment: Optional.
+    """
+
+    code: str
+    text: str
+    type: str
+    unit: str
+    comment: str
+
+
+class CommentItem(TypedDict):
+    """A single comment item for a value in API response data.
+
+    Keys:
+        variable: Identifies the column.
+        value: Identifies the value.
+        comment: A comment for the specified value.
+    """
+
+    variable: str
+    value: str
+    comment: str
+
+
+class DataPoint(TypedDict):
+    """A single data point from the API response data.
+
+    All keys and values should be in same order as "columns" items in API response data.
+
+    Keys:
+        key: List of all dimension and time column values.
+        values: List of all measure column values.
+    """
+
+    key: list[str]
+    values: list[str]
+
+
+class MetadataItem(TypedDict):
+    """Metadata details for the database table that the response is based on."""
+
+    updated: str
+    label: str
+    source: str
+
+
+class ResponseData(TypedDict):
+    """The entire data part of the API response."""
+
+    columns: list[ColumnItem]
+    comments: list[CommentItem]
+    data: list[DataPoint]
+    metadata: list[MetadataItem]
+
+
+class ResponseDataError(Exception):
+    """When the data from API response might cause the program to malfunction."""
+
+    pass
+
+
+class Command(BaseCommand):
+    help = "Import indexes related to Level Check (Tasotarkistus) from Tilastokeskus database."
+
+    def handle(self, *args, **options):
+        for index_input in INDEXES_TO_IMPORT:
+            self.stdout.write(f'Index: "{index_input["name"]}"')
+
+            index_data = _get_index_data(index_input["url"], index_input["code"])
+            columns = index_data.get("columns", [])
+
+            # Find the order of the necessary columns to avoid future failures
+            # if columns are added, removed, or moved.
+            try:
+                quarter_key_pos = _find_key_position(columns, "Vuosineljännes")
+                region_key_pos = _find_key_position(columns, "Alue")
+                index_column_pos = _find_column_position(columns, index_input["code"])
+                number_value_pos = _find_value_position(columns, index_input["code"])
+            except ResponseDataError as e:
+                self.stderr.write(f"Error: {e}. Skipping this index import")
+                continue
+
+            index_column_details = columns[index_column_pos]
+
+            metadata = index_data.get("metadata", [])
+            metadata = next(iter(metadata), {})
+
+            index, created = (
+                OldDwellingsInHousingCompaniesPriceIndex.objects.update_or_create(
```

Comment:
Wondering about the default values, if they don't exist is there a need to create or update anything?
The default values for `dict.get()` make it look like it would then create empty cells in data.

Reply:
The defaults are mostly empty strings because [Django reference] (https://docs.djangoproject.com/en/4.2/ref/models/fields/#django.db.models.Field.null) advises to avoid both `null=True` and `blank=True` for CharFields and TextFields in the models, to avoid possible ambiguity in having two possibilities for "no value".

But in this project I have seen that the `null` SQL value has been the default way for representing missing string values. I can align with that, if you prefer.

Reply:
What I meant was that if the values are not there, should we go ahead and update existing values with empty values?

Reply:
Ah, ok. I think my reasoning here aligns with my comment to your similar question for IndexNumber update: https://github.com/City-of-Helsinki/mvj/pull/696#discussion_r1719469218